File: Lab6/Product.java
import java.text.*;

public class Product {
    private String name;
    private int stock;
    private double price;

    public Product(String name, int stock, double price) {
        this.name = name;
        this.stock = stock;
        this.price = price;
    }

    public String getName() {
        return this.name;
    }

    public boolean isEmpty() {
        return this.stock == 0;
    }

    public boolean has(int n) {
        return this.stock >= n;
    }

    public double sell(int n) {
        if (has(n)) {
            this.stock -= n;
            return n * this.price;
        } else {
            return 0.0;
        }
    }

    public void restock(int n) {
        this.stock += n;
    }

    @Override
    public String toString() {
        return name + " - " + stock + " at $" + formatted(price); 
    }

    private String formatted(double money) {
        return new DecimalFormat("###,##0.00").format(money);
    }
}

File: Lab6/Product.py
class Product:
    def __init__(self, name, stock, price):
        self.name = name
        self.stock = stock
        self.price = price

    def is_empty(self):
        return self.stock == 0

    def has(self, n):
        return self.stock >= n

    def sell(self, n):
        if self.has(n):
            self.stock -= n
            return n * self.price
        else:
            return 0.0

    def restock(self, n):
        self.stock += n

    def __str__(self):
        return f"{self.name} - {self.stock} at ${self.price:,.2f}"

